packages/store/addon/-private/system/coerce-id.ts

```diff
@@ -25,4 +24,19 @@ function coerceId(id: Coercable): string | null {
   return '' + id;
 }
 
+export function ensureStringId(id: Coercable): string {
+  let normalized: string | null = null;
+  if (typeof id === 'string') {
+    normalized = id.length > 0 ? id : null;
+  } else if (typeof id === 'number' && !isNaN(id)) {
+    normalized = '' + id;
+  }
+
+  if (DEBUG && normalized === null) {
+    throw new Error(`Expected id to be a string or number, recieved ${String(id)}`);
+  }
+
+  return normalized as string;
+}
```

Comment:
the non-null assertion operator is a safer cast for this kind of thing.
```ts
return normalized!;
```

to see the difference, look at the types of x and y below

```ts
function foo(): string | number | boolean | null {
  return '';
}
const x = foo()!;
const y = foo() as string;
```

Reply:
I see the value more generally, in this case it should only ever be string but will make the update